general/development/faq.md

```diff
@@ -0,0 +1,110 @@
+
+Because of Moodle's modular nature, there is no single, detailed representation of the full database schema. Instead, the tables for each part of Moodle are defined in a database-neutral XML format in each part of Moodle (see [XMLDB](https://docs.moodle.org/19/en/Database_FAQ#XMLDB)). Look for files called `install.xml` located in folders called `db` throughout your Moodle installation.
+
+Alternatively, log in as an administrator on your Moodle site and go to _Site administration -> Development -> XMLDB editor_. Use the _Doc_ link to see automatically generated documentation built from `install.xml` files.
+
+See also [Database FAQ](https://docs.moodle.org/19/en/Database_FAQ).
+
+### How do I find out the currently-logged-on user?
+
+The global object `$USER` contains various properties pertaining to the currently logged in user. Here are just a few examples.
+
+- First name `$USER->firstname`
+- Last name `$USER->lastname`
+- User ID `$USER->id`
+
+### How do I find out the current course?
+
+Just like the global object `$USER`, there is a global object called `$COURSE` which contains details about a course.
+
+- Course ID `$COURSE->id`
+- Course name `$COURSE->fullname`
+
+### How do I insert/retrieve records in the database?
+
+Use the global `$DB` object, as described in the [Data manipulation API](/docs/apis/core/dml) documentation, to insert and retrieve data without having to create your own database connection.
+
+### How do I get/set configuration settings?
+
+#### config table
+
+To get config values you would typically access the global `$CFG` object directly, which is automatically created by the core Moodle scripts. To set these **main** config values use `set_config($name, $value)`. The values are stored in the Moodle _config_ database table, but these functions take care of cacheing on your behalf, so you should always use these rather than fetching the records directly.
+
+#### config_plugin table
+
+There is also a second table of config settings specifically for plugins called _config_plugin_. These are not automatically loaded into the `$CFG` object. To get these you would use `get_config($plugin, $name)`. To set them use `set_config($name, $value, $plugin)`.
+
+On top of those global configuration values, individual blocks may also have a configuration object associated with it (the data is serialised and stored in the _block_instance_ table). Within blocks, this data is automatically loaded into the _config_ attribute of the block.
+
+### How do I migrate my Moodle site?
+
+There may be times when you need to move your Moodle site from one server to another. Find out all about [migrating your Moodle site](https://docs.moodle.org/400/en/Moodle_migration).
+
+## Learn more
+
+Two online courses specifically for Moodle developers and are run by [HRDNZ](http://www.hrdnz.com) (certified Moodle Partner since 2006) specifically helping new Moodle developers get up-to-speed with Moodle development.
```

Comment:
Instead of pointing to these courses, I think it would be better to reference the Developer pathway in Moodle Academy: https://moodle.academy/course/index.php?categoryid=4